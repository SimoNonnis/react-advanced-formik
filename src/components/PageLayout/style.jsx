import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { H1, Button } from '@housesimple/react-components';

export const PageOuter = styled.div`
  background-color: #f9f9f9;
  padding: ${({ theme: { spacing } }) => `${spacing.sm} ${spacing.md}`};
  max-width: 770px;
  margin: 0 auto;
`;

export const PageInner = styled.div`
  padding-top: ${({ theme: { spacing } }) => `${spacing.lg}`};
`;

export const BackLinkSection = styled.div`
  text-align: right;
`;

export const BackLink = styled(Link)`
  display: inline-block;
  text-decoration: none;
  color: ${({ theme }) => theme.color.mediumBlue};
  font-size: 16px;
`;

export const LegacyBack = styled(Button)`
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`;

export const Arrow = styled.div`
  display: inline-block;

  &:before,
  &:after {
    position: relative;
    content: '';
    display: inline-block;
    vertical-align: middle;
    bottom: 1px;
  }

  &:before {
    border-color: ${({ theme }) => theme.color.mediumBlue};
    border-style: solid;
    border-width: 1px 1px 0 0;
    height: 8px;
    width: 8px;
    left: 1px;
    transform: rotate(-135deg);
  }

  &:after {
    border-radius: 50% 0 0 50%;
    height: 1px;
    background-color: ${({ theme }) => theme.color.mediumBlue};
    width: 16px;
    left: -8.5px;
  }
`;

export const Title = styled(H1)`
  margin-bottom: ${({ theme }) => theme.spacing.md};
`;
