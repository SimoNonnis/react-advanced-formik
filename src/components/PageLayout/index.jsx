import React from 'react';
import PropTypes from 'prop-types';
import { PageOuter, PageInner, BackLinkSection, Title, LegacyBack, BackLink, Arrow } from './style';

const propTypes = {
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]).isRequired,
  back: PropTypes.string,
  title: PropTypes.string,
  legacy: PropTypes.bool,
};

const defaultProps = {
  back: null,
  title: null,
  legacy: false,
};

const onLegacyBack = () => {
  window.history.back();
};

const PageLayout = ({ children, legacy, back, title }) => {
  const hasBack = legacy || back;
  const backLink = legacy ? (
    <LegacyBack type="link" onClick={onLegacyBack}>
      <Arrow /> Back
    </LegacyBack>
  ) : (
    <BackLink to={back}>
      <Arrow /> Back
    </BackLink>
  );

  return (
    <PageOuter>
      {hasBack && <BackLinkSection>{backLink}</BackLinkSection>}
      <PageInner>
        {title && <Title>{title}</Title>}
        {children}
      </PageInner>
    </PageOuter>
  );
};

PageLayout.propTypes = propTypes;
PageLayout.defaultProps = defaultProps;

export default PageLayout;
