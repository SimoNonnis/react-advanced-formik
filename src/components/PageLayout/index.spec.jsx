/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import PageLayout from './';

const children = () => <div>This is test children</div>;

describe('Test <PageLayout /> component', () => {
  it('should render PageLayout correctly with required props', () => {
    const wrapper = shallow(<PageLayout>{children}</PageLayout>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PageLayout correctly with back props', () => {
    const wrapper = shallow(<PageLayout back="/test">{children}</PageLayout>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render PageLayout correctly with title props', () => {
    const wrapper = shallow(<PageLayout title="Title">{children}</PageLayout>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
