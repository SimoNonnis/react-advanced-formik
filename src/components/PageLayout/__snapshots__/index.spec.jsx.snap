// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PageLayout /> component should render PageLayout correctly with back props 1`] = `
<styled.div>
  <styled.div>
    <Styled(Link)
      to="/test"
    >
      <styled.div />
       Back
    </Styled(Link)>
  </styled.div>
  <styled.div />
</styled.div>
`;

exports[`Test <PageLayout /> component should render PageLayout correctly with required props 1`] = `
<styled.div>
  <styled.div />
</styled.div>
`;

exports[`Test <PageLayout /> component should render PageLayout correctly with title props 1`] = `
<styled.div>
  <styled.div>
    <Styled(H1)>
      Title
    </Styled(H1)>
  </styled.div>
</styled.div>
`;
